src/spring-cloud/azext_spring_cloud/_utils.py

```diff
@@ -14,15 +14,16 @@
 from json import dumps
 from knack.util import CLIError, todict
 from knack.log import get_logger
+from .vendored_sdks.appplatform.models import _app_platform_management_client_enums as AppPlatformEnums
 from ._client_factory import cf_resource_groups
 
 
 logger = get_logger(__name__)
 
 
-def _get_upload_local_file(jar_path=None):
-    file_path = jar_path
-    file_type = "Jar"
+def _get_upload_local_file(runtime_version, artifact_path=None):
+    file_path = artifact_path
+    file_type = "NetCoreZip" if runtime_version == AppPlatformEnums.RuntimeVersion.net_core_31 else "Jar"
 
     if file_path is None:
```

Comment:
Do you need to check file_type != NetCoreZip here?

Reply:
Don't quite understand the question here. 3 different scenarios:

1. source deployment: file_type = jar and file_path is None
2. jar deployment: file_type =jar and file_path is not none
3. netcorezip deployment: file_type=netcorezip and file_path is not none

Reply:
I see, I didn't realize the file_path is not required only when source deployment. The code miss led me to consider the un-existing scenario file_path w/ file_type = netcorezip. Would you mind commenting your scenraios in the code?